packages/common/src/types.ts

```diff
@@ -27,6 +27,10 @@ export interface Chain {
   consensus?: {
     type: string
     algorithm: string
+    clique: {
+      period: number
+      epoch: number
+    }
```

Comment:
This was too strict and breaks typing in non-PoA contexts, have fixed this in 6fe4a1ddab7c78a5551dce139de4b3e988f2cd7b.

Reply:
ok great, I was wondering, it might be helpful if all the options were optional, at least for my use in `forCustomChain` , I only wanted to set `epoch` to `3`, but I had to provide all the other values:

https://github.com/ethereumjs/ethereumjs-monorepo/blob/6721321fb46e8543eeb1027d11fdd8cdf8841994/packages/blockchain/test/clique.spec.ts#L535-L548

I understand this is a limitation of how the code is set up in `forCustomChain` though, since it's using the spread operator, it does not allow for deep nesting replacement of single values, it would replace the whole block provided. Not sure if there is a better technique we could use.

Reply:
Wouldn't be a fan if we loosen this here in general since the type safety of these chain and HF definition JSON files is so underdeveloped already and should rather be improved towards the other side at some PiT. I get the point though. Allow `any` here for the function signature as the most simple/naive solution? 😜  Not sure, if there is something better, this should of course be preferred.

Reply:
yeah, agree, in general would love some more type safety in common coming from the json files.

what I meant though is I would love to do this:

```
 const common = Common.forCustomChain( 
   'rinkeby',
   { consensus: { clique: { epoch: 3 } } }, 
   'chainstart' 
 )
```

however the current way the code is set up, then` { consensus: type, algorithm, clique: { period } }` would all be undefined. we would need some more sophisticated logic for deeply replacing nested values, rather than the simple solution of using the spread operator ([here](https://github.com/ethereumjs/ethereumjs-monorepo/blob/55f640003c9488fe9285d2c86b0c49bce643d0b5/packages/common/src/index.ts#L89)).

Reply:
Ah ok, thanks for the clarification.